lib/castle_web/controllers/api/episode_controller.ex

```diff
@@ -2,15 +2,24 @@ defmodule CastleWeb.API.EpisodeController do
   use CastleWeb, :controller
 
   alias Castle.Rollup.Data.Totals, as: Totals
+  alias Castle.Rollup.Data.Trends, as: Trends
 
   def index(conn, _params) do
     render conn, "index.json", conn: conn, episodes: Totals.episodes(), meta: %{cached: true}
   end
 
   def show(conn, %{"id" => id}) do
-    case Totals.episode(id) do
-      nil -> send_resp conn, 404, "Episode #{id} not found"
-      ep -> render conn, "show.json", conn: conn, episode: ep, meta: %{cached: true}
+    case assemble_data(id) do
+      {nil, _} ->
+        send_resp conn, 404, "Episode #{id} not found"
+      {ep, trends} ->
+        render conn, "show.json", conn: conn, episode: ep, trends: trends, meta: %{cached: true}
     end
   end
+
+  defp assemble_data(id) do
+    t1 = Task.async(fn -> Totals.episode(id) end)
+    t2 = Task.async(fn -> Trends.episode(id) end)
```

Comment:
The redis data being cached includes _all_ episode guids ... maybe not the smallest payload ever.  So better to fetch them in parallel.

(If I was doing this again, I'd probably try using a redis hash instead of just a big json blob)

Reply:
https://hexdocs.pm/elixir/Task.html 
-> cool